### DISCUSSION TITLE
spi cs-gpios issue (GPIO_ACTIVE_LOW does not work)

### DISCUSSION THREAD

**Discussion by 2songh** [zephyrproject-rtos/zephyr#67185] [Q&A]
zephyr version: v2.4
my board: STM32l433
spi device: microchip,mcp23s08

*****

**[.dts file]**
![image](https://github.com/zephyrproject-rtos/zephyr/assets/126455296/43beaaeb-c461-44de-a571-8d9998a2510e)

**[prj.conf]**
![image](https://github.com/zephyrproject-rtos/zephyr/assets/126455296/0feb1d2e-c9ce-4e90-973e-c0e3fee2880c)

I want to get the input value of my device through spi communication.
But my cs-gpios only works with GPIO_ACTIVE_HIGH.
I hope it works with GPIO_ACTIVE_LOW.

*****

![image](https://github.com/zephyrproject-rtos/zephyr/assets/126455296/c1b1515b-abd1-4466-b5e3-a1270d48b2f5)
(yellow: Chip Select / blue: Clock)

Is there a problem configuring my dts file or prj.conf file?
Additionally, Should I  suspect a board configuration or gpio driver issue?

Any opinions are welcome, so please give me your advice.

**Comment by ndrs-pst** [comment]
@2songh, Since your Zephyr version is outdated, please use a currently supported version, such as v2.7.5 or
v3.5.99, along with the provided dts.
By the way, since cs-gpios are managed by spi_ll_stm32.c and shared among various SPI interface ICs, I believe it should be okay.
```

#include <freq.h>                           /* DT_FREQ_M */
&spi1 {
    status = "okay";
    cs-gpios = <&gpiob 12 GPIO_ACTIVE_LOW>;

    msp23s08: mcp23s08@0 {
        compatible = "microchip,mcp23sxx";
        reg  = <0>;
        spi-max-frequency = <DT_FREQ_M(5)>;

        gpio-controller;        /* The following descriptions are from gpio-controller.yaml */
        #gpio-cells = <2>;
        ngpios = <8>;           /* 8-bit (pins) */
        int-gpios = <&gpiob 22 GPIO_ACTIVE_LOW>;    /* optional, please adjust with your board or comment out */
        reset-gpios = <&gpiob 17 GPIO_ACTIVE_LOW>;  /* optional, please adjust with your board or comment out */
        status = "okay";
    };
```

**Reply by 2songh** [reply]
Thanks for your answer.

We created driver mcp23s08 by referencing mcp23s17 in version 2.2.99.
mcp23s08 worked normally in that version.
(dts, prj.conf, pinmux, defconfig are all the same.)

**Additionally, normal operation was achieved when tested using mcp23sxx in version 3.5.**

Are we correct in seeing this as a problem with version 2.4?
We hope to use mcp23s08 in version 2.4.
Are there any more files or methods to check?

**Comment by ndrs-pst** [comment]
@2songh, Understood. If you're still looking to use the Zephyr 2.x version, I recommend opting for version 2.7.5.
Identifying regressions and managing backports might be more challenging in this case.
